launcher/LaunchController.cpp

```diff
@@ -116,6 +116,12 @@ void LaunchController::login() {
         m_session->wants_online = m_online;
         m_accountToUse->fillSession(m_session);
 
+        // Launch immediately in true offline mode
```

Comment:
Does this run if the account is "expired" and you have to re-login?

Reply:
This only runs for the offline mode accounts, which never expire and never need to re-login.